src/test/java/org/ow2/proactive/connector/iaas/cloud/provider/azure/AzureProviderTest.java

```diff
@@ -0,0 +1,956 @@
+                                                      any(Region.class),
+                                                      any(ResourceGroup.class),
+                                                      anyString(),
+                                                      anyString())).thenReturn(creatableVirtualNetwork);
+        when(azureProviderUtils.preparePublicIPAddress(any(Azure.class),
+                                                       any(Region.class),
+                                                       any(ResourceGroup.class),
+                                                       anyString(),
+                                                       anyBoolean())).thenReturn(creatablePublicIpAddress);
+        when(azureProviderUtils.prepareNetworkInterfaceFromScratch(any(Azure.class),
+                                                                   any(Region.class),
+                                                                   any(ResourceGroup.class),
+                                                                   anyString(),
+                                                                   any(Creatable.class),
+                                                                   any(Creatable.class),
+                                                                   any(Creatable.class))).thenReturn(creatableNetworkInterface);
+        when(azureProviderUtils.prepareSSHNetworkSecurityGroup(any(Azure.class),
+                                                               any(Region.class),
+                                                               any(ResourceGroup.class),
+                                                               anyString())).thenReturn(creatableNetworkSecurityGroup);
+
+        // Images
+        PagedList<VirtualMachineCustomImage> pagedListCustomImage = new PagedList<VirtualMachineCustomImage>() {
+            @Override
+            public Page<VirtualMachineCustomImage> nextPage(String nextPageLink) throws RestException, IOException {
+                return null;
+            }
+        };
+        pagedListCustomImage.add(virtualMachineCustomImage);
+        when(virtualMachineCustomImages.list()).thenReturn(pagedListCustomImage);
+        when(azureService.virtualMachineCustomImages()).thenReturn(virtualMachineCustomImages);
+
+        // VirtualMachines
+        PagedList<VirtualMachine> pagedListVirtualMachine = new PagedList<VirtualMachine>() {
+            @Override
+            public Page<VirtualMachine> nextPage(String nextPageLink) throws RestException, IOException {
+                return null;
+            }
+        };
+        pagedListVirtualMachine.add(virtualMachine);
+        when(virtualMachines.list()).thenReturn(pagedListVirtualMachine);
+        when(azureService.virtualMachines()).thenReturn(virtualMachines);
+
+        // ResourceGroups
+        PagedList<ResourceGroup> pagedListResourceGroup = new PagedList<ResourceGroup>() {
```

Comment:
It seems that this `PagedList` construction is duplicated 9 times. You could factorize as follows:

```
private <T> PagedList<T> getVirtualMachineCustomImages() {
        return new PagedList<T>() {
            @Override
            public Page<T> nextPage(String nextPageLink) throws RestException, IOException {
                return null;
            }
        };
    }
```